### DISCUSSION TITLE
Encountering Error with Doctrine: "Doctrine\ORM\ORMInvalidArgumentException: A new entity was found through the relationship".

### DISCUSSION THREAD

**Discussion by thienhuynhtps** [symfony/symfony#58730] [Q&A]
Hi. Can anyone help me ?

**#Overview:**

I am currently writing a feature for users to create shops. 
After a user creates a shop, it will require admin approval. The admin will then review and approve it.
A shop contains various information and allows image uploads.

**#Environment:**

- Symfony 5.4.45
- MySQL
- PHP 8.1.20
- "doctrine/dbal": "3.9.3"
- "doctrine/doctrine-bundle": "2.11.1"

**#Flow Code:**

When the admin approves, the information will be transferred from the temporary table to the main table.

**#Issues:**

The previous code was functioning normally for a long time and has not had any related updates.

However, today an error has occurred, affecting all shops when approval.

**#Error message:** 

```
Doctrine\ORM\ORMInvalidArgumentException: A new entity was found through the relationship \app\Entity\Shop#ShopImage that was not configured to cascade persist operations for entity: app\Entity\Shop\ShopImage@7139. 
To solve this issue: Either explicitly call EntityManager#persist() on this unknown entity or configure cascade persist this association in the mapping, for example, @ManyToOne(..,cascade={"persist"}).
```

**One notable point is that shops without uploaded images are still encountering this error.**

**#Code Causing the Error:**

```
$this->filterImageNeedToDeleteS3($ShopTmp->getShopImageTmp(), $Shop->getShopImage());
$this->deleteShopImages($Shop);

foreach ($ShopTmp->getShopImageTmp() as $ShopImageTmp) {
      if ($ShopImageTmp->getFileName()) {
              $ShopImage = new ShopImage();
              $ShopImage->setShop($Shop);
              $ShopImage->setFileName($ShopImageTmp->getFileName());
              $this->entityManager->persist($ShopImage);
              $this->entityManager->flush();
      }
}

if (count($ShopTmp->getShopImageTmp()) < 1) {
        $ShopImage = new ShopImage();
        $ShopImage->setShop($Shop);
        $ShopImage->setFileName(null);
        $this->entityManager->persist($ShopImage);
        $this->entityManager->flush();
}
```

**#Things I Tried to Fix:**

1. Added cascade={"persist"} to the entity as per the error message.
=> When I added it, the save operation was successful, but now there are two identical records saved in the database.
```

 /**
   * @ORM\ManyToOne(targetEntity="app\Entity\Shop", inversedBy="ShopImage", cascade={"persist"})
   * @ORM\JoinColumn(name="shop_id", referencedColumnName="id", onDelete="CASCADE")
   * @ORM\OrderBy({
   *     "sort_no"="ASC"
   * })
   */
   private $Shop;
```

2. Later, I discovered that the code for deleting the old images using EntityManager no longer works as it did before. 
I had to switch to deleting using QueryBuilder.

$this->deleteShopImages($Shop);

```
// Old code

$ShopImages = $this->shopImageRepository->findBy([
            'Shop' => $Shop,
]);

 foreach ($ShopImages as $ShopImage) {
      $this->entityManager->remove($ShopImage);
 }
```
```
// New code
 $this->shopImageRepository
         ->createQueryBuilder('s')
         ->delete()
         ->where("s.Shop= :Shop")
         ->setParameter('Shop', $Shop)
         ->getQuery()
         ->execute()
 ;
```

3. After all, I accepted the situation of saving two records and will filter them out on the UI.

At this point, I still don’t understand the cause. 
I suspect that it might be due to my system rebuilding every time code is updated through CI/CD, and each build reinstalls Composer.
Could it be that some Doctrine library packages have been updated, leading to this issue?

Could everyone provide me with some suggestions for what to check? Thank you!

**Comment by claudio-cesar** [comment]
Hey, how's going ?

I'm also facing this problem, but mine is a bit strange because this error only happens during automated E2E ( Cypress ) tests. During manual regression tests everything works.

Have you found the solution ?
